### PR TITLE
Test CMake builds in CI

### PR DESCRIPTION
**Pull request by o01eg**
Backport of https://github.com/godotengine/godot-cpp/pull/518 to `3.x`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -43,6 +43,66 @@ jobs:
...
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update -qq
+          sudo apt-get install -qqq build-essential pkg-config cmake
+          curl -LO https://downloads.tuxfamily.org/godotengine/3.4/Godot_v3.4-stable_linux_server.64.zip
```
Comment (Calinou) [inline .github/workflows/ci.yml:59]: Shouldn't we use 3.4.2 instead?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -43,6 +43,66 @@ jobs:
...
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update -qq
+          sudo apt-get install -qqq build-essential pkg-config cmake ninja-build
+          curl -LO https://downloads.tuxfamily.org/godotengine/3.4/Godot_v3.4-stable_linux_server.64.zip
```
Comment (Calinou) [inline .github/workflows/ci.yml:89]: Same here.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (o01eg) [reply .github/workflows/ci.yml:59]: I just use same version scons CI uses. I could try to update although.

Comment (o01eg) [reply .github/workflows/ci.yml]: @Calinou Looks like it works.

**Comment by aaronfranke** [comment]
@o01eg This PR has conflicts, it needs a rebase before it can be considered.

**Comment by o01eg** [comment]
CI workflow was drastically rewritten in those two years so this PR has no sense anymore.

**Pull request closed without merging**